src/app/CommandSender.cpp

```diff
@@ -80,24 +78,33 @@ CHIP_ERROR CommandSender::TestOnlyCommandSenderTimedRequestFlagWithNoTimedInvoke
 
     mExchangeCtx->SetResponseTimeout(timeout.ValueOr(session->ComputeRoundTripTimeout(app::kExpectedIMProcessingTime)));
 
+    if (sendTimedRequestAction && !mTimedInvokeTimeoutMs.HasValue())
+    {
+        ChipLogError(DataManagement, "Timed request action requested with no mTimedInteractionTimeoutMs");
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    if (sendTimedRequestAction)
+    {
+        ReturnErrorOnFailure(TimedRequest::Send(mExchangeCtx.Get(), mTimedInvokeTimeoutMs.Value()));
+        MoveToState(State::AwaitingTimedStatus);
+        return CHIP_NO_ERROR;
+    }
+
     return SendInvokeRequest();
 }
+
+#if CONFIG_BUILD_FOR_HOST_UNIT_TEST
+CHIP_ERROR CommandSender::TestOnlyCommandSenderTimedRequestFlagWithNoTimedInvoke(const SessionHandle & session,
+                                                                                 Optional<System::Clock::Timeout> timeout)
+{
+    VerifyOrReturnError(mTimedRequest, CHIP_ERROR_INCORRECT_STATE);
+    return SendCommandRequestInternal(session, timeout, false);
```

Comment:
Wouldn't mTimedInvokeTimeoutMs.HasValue() always be false here?  If not, perhaps document why not?  But if it would be, then we can just have `SendCommandRequestInternal` check that.

Reply:
Removed that parameter in the latest commit.